examples/rvv_strcmp.c

```diff
@@ -4,24 +4,23 @@
 
 // reference https://github.com/riscv/riscv-v-spec/blob/master/example/strcmp.s
 int strcmp_vec(const char *src1, const char *src2) {
+  const unsigned char *s1 = (const void*)src1, *s2 = (const void*)src2;
   size_t vlmax = __riscv_vsetvlmax_e8m2();
   long first_set_bit = -1;
-  size_t vl, vl1;
-  while (first_set_bit < 0) {
-    vint8m2_t vec_src1 = __riscv_vle8ff_v_i8m2(src1, &vl, vlmax);
-    vint8m2_t vec_src2 = __riscv_vle8ff_v_i8m2(src2, &vl1, vlmax);
+  size_t vl;
+  for (; first_set_bit < 0; s1 += vl, s2 += vl) {
+    vuint8m2_t vec_src1 = __riscv_vle8ff_v_u8m2(s1, &vl, vlmax);
+    vuint8m2_t vec_src2 = __riscv_vle8ff_v_u8m2(s2, &vl, vl);
 
-    vbool4_t string_terminate = __riscv_vmseq_vx_i8m2_b4(vec_src1, 0, vl);
-    vbool4_t no_equal = __riscv_vmsne_vv_i8m2_b4(vec_src1, vec_src2, vl);
+    vbool4_t string_terminate = __riscv_vmseq_vx_u8m2_b4(vec_src1, 0, vl);
+    vbool4_t no_equal = __riscv_vmsne_vv_u8m2_b4(vec_src1, vec_src2, vl);
     vbool4_t vec_terminate = __riscv_vmor_mm_b4(string_terminate, no_equal, vl);
 
     first_set_bit = __riscv_vfirst_m_b4(vec_terminate, vl);
-    src1 += vl;
-    src2 += vl;
   }
-  src1 -= vl - first_set_bit;
-  src2 -= vl - first_set_bit;
-  return *src1 - *src2;
+  s1 -= vl - first_set_bit;
+  s2 -= vl - first_set_bit;
```

Comment:
Does this fix the problem mentioned in #244?

Reply:
Changing `__riscv_vle8ff_v_i8m2(src2, &vl1, vlmax)` to `__riscv_vle8ff_v_u8m2(s2, &vl, vl);`  on line 13 should have fixed it.

Reply:
I see the fix now. Thank you for explaining.